### PR TITLE
CGMES. Fix import of condensers

### PR DESCRIPTION
**Pull request by zamarrenolm**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

- In general, for all CIM versions, the `condenser` flag of the `Generator` objects was not set for synchronous machines of type (kind) condensers.
- In CIM14 synchronous machines that were only condensers were not handled properly.

**What is the new behavior (if this is a feature change)?**
- The `condenser` flag is set in the converged `Generators` if the type/kind of the synchronous machine is a condenser (and only a condenser).
- CIM14 synchronous machines of type `condenser` are handled as condensers.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

**Other information**:

Bad handling of condensers in CIM14 was raised in #3214.

### CODE REVIEW

**Review by rcourtier — APPROVED** [review]

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/SynchronousMachineConversion.java [hunk 1]
```diff
@@ -61,6 +62,7 @@ public void convert() {
                 .setRatedS(ratedS);
         identify(adder);
         connect(adder);
+        adder.setCondenser(isCondenser);
```
[the remark this answers is not in the document]
Comment (olperr1) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/SynchronousMachineConversion.java:65]: After a second thought, we can keep it like this. I will just update the name/description to highlight the fact that this PR concerns all CIM versions.

**Review by olperr1 — APPROVED** [review]

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java [hunk 2]
```diff
@@ -35,6 +35,19 @@ void cgmes2condenserConversionTest() {
...
         assertEquals(0, g.getMaxP());
+        assertTrue(g.isCondenser());
+    }
+
+    @Test
+    void cgmes2condenserExportTest() throws IOException {
+        Network network = Network.read("condenser2_EQ.xml", getClass().getResourceAsStream("/issues/condenser2_EQ.xml"));
+        String basename = "condenser2";
+        network.write("CGMES", null, tmpDir.resolve(basename));
+        String eq = Files.readString(tmpDir.resolve(basename + "_EQ.xml"));
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java:46]: Can be simplified as follow:
`String eq = ConversionUtil.writeCgmesProfile(network, "EQ", tmpDir);`

**Review by rcourtier — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java [hunk 3]
```diff
@@ -35,6 +34,19 @@ void cgmes2condenserConversionTest() {
...
         assertEquals(0, g.getMinP());
         assertEquals(0, g.getMaxP());
+        assertTrue(g.isCondenser());
+    }
+
+    @Test
+    void cgmes2condenserExportTest() throws IOException {
+        Network network = Network.read("condenser2_EQ.xml", getClass().getResourceAsStream("/issues/condenser2_EQ.xml"));
+        String basename = "condenser2";
+        network.write("CGMES", null, tmpDir.resolve(basename));
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java:44]: You can remove that.

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java [hunk 4]
```diff
@@ -44,17 +56,19 @@ void cgmes3condenserConversionTest() {
...
         assertEquals(0, g.getMinP());
         assertEquals(0, g.getMaxP());
+        assertTrue(g.isCondenser());
     }
 
     @Test
     void cgmes3condenserExportTest() throws IOException {
         Network network = Network.read("condenser3_EQ.xml", getClass().getResourceAsStream("/issues/condenser3_EQ.xml"));
         String basename = "condenser3";
         network.write("CGMES", null, tmpDir.resolve(basename));
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java:66]: You can remove that.

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java [hunk 5]
```diff
@@ -68,11 +82,12 @@ void cgmes3condenserReferencePriorityTest() throws IOException {
         assertEquals(g.getTerminal(), referencePriority.getTerminal());
 
         network.write("CGMES", null, tmpDir.resolve(basename));
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/CondenserImportExportTest.java:84]: You can remove that.

**Review by rcourtier — CHANGES_REQUESTED** [review]

**Review by rcourtier — APPROVED** [review]

**Pull request merged**